### PR TITLE
Add new/missing content to general docs -  Closes #1417

### LINKED ISSUE #1417: Add new/missing content to general docs
### Description

Add missing content:

- overview pages
- api page

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1417

### CODE REVIEW

File: docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc [hunk 1]
```diff
@@ -0,0 +1,225 @@
+= Publish/Subscribe API (Lisk Sevice)
+:toc:
+
+The Publish/Subscribe API is sometimes also called Event-Driven API.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc]:
```suggestion
The Publish/Subscribe API is sometimes also called an Event-Driven API.
```

File: docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+The biggest difference between Event-Driven and regular REST API is not only technical.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc]:
```suggestion
The main difference between Event-Driven and regular REST API is not only technical.
```

File: docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+In practice, a two-way streaming connection is used, which means that not only can the client request the server for a data update, (and also potential updates) but also the server can notify the client about new data instantly as it arrives.
+
+Lisk Service leverages the two-way communication approach by utilizing the WebSocket library responsible for updating users about changes in the blockchain network and markets.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc]:
```suggestion
Lisk Service leverages the two-way communication approach by utilizing the WebSocket library responsible for updating users regarding changes in the blockchain network and markets.
```

File: docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+If you are interested in accessing the testnet network, this can be accomplished by using the `wss://testnet-service.lisk.com/blockchain` endpoint.
+
+Important: The Lisk Service WebSocket API uses the `socket.io` library.
+This implementation is compatible with the version 2.0 of the socket.io library.
+Using the wrong major version might result in a broken connection and messages not being passed.
+
+The specification below contains numerous examples how to use the API in practice.
+
+== Endpoint logic
+The logic of the endpoints is derived as follows: the method naming is always based on the following pattern: `<action>.<entity>`, where the `action` is equivalent to method type performed on the server (ie. `update`) and `entity` is a part of the application logic, ex. `accounts`, `transactions`.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc]:
```suggestion
The logic of the endpoints is derived as follows: the method naming is always based on the following pattern: `<action>.<entity>`, where the `action` is equivalent to the method type performed on the server (i.e., `update`), and `entity` is a part of the application logic, ex. `accounts`, `transactions`.
```

File: docs/modules/ROOT/pages/api/lisk-service-rpc.adoc [hunk 5]
```diff
@@ -0,0 +1,1289 @@
...
+=== Access paths and compatibility
+The WebSocket API can be accessed by the `wss://service.lisk.com/rpc-v2`.
+
+The testnet network can also be accessed by `wss://testnet-service.lisk.com/rpc-v2`.
+
+The Lisk Service WebSocket API uses the `socket.io` library and is compatible with the JSON-RPC 2.0 standard.
+The specification below contains numerous examples how to use the API in practice.
+
+=== Endpoint logic
+The logic of the endpoints is derived as follows: the method naming is always based on the following pattern: `<action>.<entity>`, where the action is equivalent to HTTP standard (GET, POST, PUT, etc.) and `entity` is a part of the application logic, ex. `accounts`, transactions and so on.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-rpc.adoc]:
```suggestion
The logic of the endpoints is derived as follows: the method naming is always based on the following pattern: `<action>.<entity>`, where the action is equivalent to the  HTTP standard (GET, POST, PUT, etc.), and `entity` is a part of the application logic, ex. `accounts`, transactions and so on.
```

File: docs/modules/ROOT/pages/api/lisk-service-rpc.adoc [hunk 6]
```diff
@@ -0,0 +1,1289 @@
...
+    "result": {
+        "data": {}, // Contains the requested data
+        "meta": {}, // Contains additional metadata, e.g. the values of `limit` and `offset`
+    },
+    "id": 1    // Number of response in chain
+}
+----
+
+== Date Format
+In the contrary to the original Lisk Core API, all timestamps used by the Lisk Service are in the UNIX timestamp format.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-rpc.adoc]:
```suggestion
Contrary to the original Lisk Core API, all timestamps used by the Lisk Service are in the UNIX timestamp format.
```

File: docs/modules/ROOT/pages/api/lisk-service-rpc.adoc [hunk 7]
```diff
@@ -0,0 +1,1289 @@
...
+    "jsonrpc": "2.0",
+    "method": "get.accounts",
+    "params": {
+        "address": "lskzkfw7ofgp3uusknbetemrey4aeatgf2ntbhcds"
+    }
+}
+----
+
+==== get.votes_sent
+Retrieves votes of a single account based on address, public key or delegate name.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-rpc.adoc]:
```suggestion
Retrieves votes of a single account based on address, public key, or delegate name.
```

File: docs/modules/ROOT/pages/api/lisk-service-rpc.adoc [hunk 8]
```diff
@@ -0,0 +1,1289 @@
...
+{
+    "method": "get.votes_sent",
+    "params": {
+        "address": "lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu"
+    }
+}
+----
+
+==== get.votes_received
+Retrieves voters for a delegate account based on address, public key or delegate name.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-rpc.adoc]:
```suggestion
Retrieves voters for a delegate account based on address, public key, or delegate name.
```

File: docs/modules/ROOT/pages/api/lisk-service-rpc.adoc [hunk 9]
```diff
@@ -0,0 +1,1289 @@
...
+    "method": "get.votes_received",
+    "params": {
+        "address": "lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu"
+    }
+}
+----
+
+=== Blocks
+==== get.blocks
+Retrieves blocks from the blockchain based on ID, height or account.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-rpc.adoc]:
```suggestion
Retrieves blocks from the blockchain based on ID, height, or account.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 10]
```diff
@@ -1 +1,48 @@
...
+
+The "Build Blockchain" category contains all the guides related to the development of a blockchain application.
+
+== Quickstart
+
+The xref:{url_guide_quickstart}[] guide describes in a few simple steps how to set up a blockchain application and how to start it successfully for the first time.
+
+== The Hello World application
+
+In the following guides it is explained step by step how to create a very simple blockchain application by the example of the "Hello World" blockchain application, which allows users to save hello messages in their accounts.
```
Comment (chris529) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
In the following guides, it is explained step by step how to create a very simple blockchain application by using the example of the "Hello World" blockchain application, which allows users to save hello messages in their accounts.
```

File: docs/modules/ROOT/pages/integrate-blockchain/decoding-transactions-blocks.adoc [hunk 11]
```diff
@@ -17,13 +17,15 @@ Mona Bärenfänger <mona@lightcurve.io>
...
+Blocks and transactions can be decoded conveniently with the xref:{url_references_apiclient}[apiClient] library of the Lisk SDK.
 
 == Connecting a node with the API client
 
-Before the API client can be used, it needs to connect to a blockchain node. This can be achieved either via a WebSocket (on the local and remote server), or an IPC (only on the local server) connection.
+Before the API client can be used, it needs to connect to a blockchain node.
+This can be achieved either via a WebSocket (local or remote), or an IPC (only on the local server) connection.
 
-In most scenarios it is beneficial to have a local node running (for example the {url_example_helloapp}[Hello World application^]) to which the client can send the API requests. The RPC options in the config should be enabled and configured accordingly, see xref:{url_guides_config}[RPC configuration of a node].
+In most scenarios it is beneficial to have a local node running (for example the {url_example_helloapp}[Hello World application^]) to which the client can send the API requests.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/decoding-transactions-blocks.adoc]:
```suggestion
In most scenarios, it is beneficial to have a local node running (for example, the {url_example_helloapp}[Hello World application^]) to which the client can send the API requests.
```

**Review by chris529 — COMMENTED** [review]
Looks good, just a few very minor points 👍

File: docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc [hunk 12]
```diff
@@ -0,0 +1,225 @@
+= Publish/Subscribe API (Lisk Sevice)
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-service-pubsub.adoc]:
```suggestion
= Publish/Subscribe API (Lisk Service)
```

**Review by chris529 — COMMENTED** [review]
Looks great! Only 1 very minor point 👍

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 13]
```diff
@@ -1 +1,49 @@
...
+
+== Quickstart
+
+The xref:{url_guide_quickstart}[] guide describes in a few simple steps how to set up a blockchain application and how to start it successfully for the first time.
+
+== The Hello World application
+
+In the following guides, it is explained step by step how to create a very simple blockchain application by using the example of the "Hello World" blockchain application, which allows users to save hello messages in their accounts.
+
+The following image provides an overview of the "Hello" application:
```
Comment (shuse2) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]: it feels little bit out of place to have hello world application here

File: docs/modules/ROOT/pages/api/lisk-service-http-testnet.adoc [hunk 14]
```diff
@@ -1,4 +1,4 @@
-= Lisk Service HTTP API (Testnet)
+= Testnet HTTP API (Lisk Service)
```
Comment (shuse2) [inline docs/modules/ROOT/pages/api/lisk-service-http-testnet.adoc:1]: Maybe we don't need testnet api if we have service doc for future version?

File: docs/modules/ROOT/pages/understand-blockchain/index.adoc [hunk 15]
```diff
@@ -35,11 +35,31 @@ Mona Bärenfänger <mona@lightcurve.io>
...
 
 ****
 The Lisk SDK is an open-source software development kit which enables developers to build scalable <<blockchain-applications>> in JavaScript.
 ****
 
+== What is a blockchain?
+
+At its core, a *blockchain* is a database shared across multiple computers.
+
+Operations on the database are called *transactions*.
```
Comment (shuse2) [inline docs/modules/ROOT/pages/understand-blockchain/index.adoc]: I think the below sentence is correct. maybe something like
`Triggers for the database operations are called *transactions*`?

File: docs/modules/ROOT/pages/understand-blockchain/index.adoc [hunk 16]
```diff
@@ -35,11 +35,31 @@ Mona Bärenfänger <mona@lightcurve.io>
...
+The data contained in a transaction specifies the operation to be performed.
+
+In an *account-based* blockchain, the validity of a new transaction is checked against a database of user accounts. An account is a portion of the database identified by the account *address*.
+This is opposed to *UTXO-based* blockchains (such as Bitcoin, see https://en.wikipedia.org/wiki/Bitcoin[Wikipedia page^]), where there is no concept of account and the validity of a transaction is based on a database of referenced past transactions.
+
+The cost of processing the transaction and updating the blockchain is specified in the transaction fee and paid using *tokens*.
+
+Transactions are shared among users connected in a *P2P network* and are grouped together into *blocks*.
```
Comment (shuse2) [inline docs/modules/ROOT/pages/understand-blockchain/index.adoc]:
```suggestion
Transactions are shared among nodes connected in a *P2P network* and are grouped together into *blocks*.
```

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 13 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc]: I changed the structure a little bit, is it better now?

Comment (shuse2) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc]: yes i think it looks better

**Review by shuse2 — APPROVED** [review]

**Pull request closed without merging**